utils/src/server/rpc_module.rs

```diff
@@ -180,6 +180,17 @@ impl Methods {
 		}
 	}
 
+	/// Test helper to call a method on the `RPC module` without having to spin a server up.
+	///
+	/// Converts the params to a stringified array for you if it's not already serialized to a sequence.
+	pub async fn test_call<T: Serialize>(&self, method: &str, params: T) -> Option<String> {
+		let params = serde_json::to_string(&params).ok().map(|json| {
+			let json = if json.starts_with('[') && json.ends_with(']') { json } else { format!("[{}]", json) };
```

Comment:
What if there's only one param provided to the call, and it's something that already serializes to an array like `Vec<u8>`?

Reply:
It would already be serialized to an array then we won' insert `[` json `]`, see if expression above :)

However, it could be weird if you put in a `Vec<_>` and expect it to be serialized as `[[my vec]]`, lemme double

Reply:
That's the case I was picturing; if there's an RPC call that accepts 1 param, a `Vec<u8>`, then I guess I couldn't use `test_call`, because serializing that would lead to params like `[1,2,3]` instead of a single param like `[[1,2,3]]`

Reply:
yepp, you are right.

I did it this way to work with both `RpcParams::parse` and `RpcParams::Sequence::next()`, but it won't work with doing weird things or that. I guess I was fooled because I used `JsonValue` in the test....

Don't remember if I had an issue with the parser to get this work.

Reply:
I think this is an chicken and egg problem it will always be an edge-case for this; so no good solution AFAIU.

Perhaps, I should revisit to try to add `into_rpc_test_module` or something from because this essentially becomes a footgun unless we do `if params == edge_case then panic!("not supported")`

Reply:
A call like `test_call` could accept something like `T: IntoRpcParams`, which is implemented for a bunch of tuple sizes (and maybe even slices and arrays) like:

```
impl <A: Serialize> IntoRpcParams for (A,) { .. }
impl <A: Serialize, B: Serialize> IntoRpcParams for (A,B) { .. }
impl <const N: usize, A: Serialize> IntoRpcParams for [A;N] { .. }
```

That way, You could call eg `.test_call("foo", (a, b, c))` or `.test_call("foo", [a])` or `.test_call("foo", (a,))` with low overhead?

(perhaps something like that exists in JsonRpsee already and I'm not being very helpful; I haven't checked! :))

Reply:
that's good idea, I think that's better

Reply:
I played around with tuples on a branch on saturday and it worked fine up to a point and then didn't. I ended up having to call some permutations with a string and some with a tuple, in the end I abandoned it because it felt like the user would have a deeper understanding of what's going on than I felt was warranted.

That said, it might work out better for you. :)